### ISSUE TITLE
Track progress on strategies

### ISSUE THREAD

**Issue by GayathriVenkatesh** [ifmeorg/ifme#1687]
# Description

For every strategy created, it would be nice to have a feature that allows the user to enter whether he has followed it that day.

For example, in the screenshot below, I have created 3 strategies. I feel it would be convenient to have an option to keep track of how many days I have followed them. 

![Screenshot from 2020-03-08 00-23-48](https://user-images.githubusercontent.com/32172764/76151184-23ea9500-60d8-11ea-89ec-1c5aed3b42f5.png)

When the user clicks on a date, it can show the list of all strategies with checkboxes. The user can then check or uncheck each of them.
There could also be a weekly report on how many days the user has achieved the targets.

---

Please assign yourself (via the Assignees dropdown), if you do want to work on this issue. Can't find yourself? You need to [join our organization](https://github.com/ifmeorg/ifme/wiki/Join-Our-Slack).

Check out our [Picking Up Issues](https://github.com/ifmeorg/ifme/wiki/Picking-Up-Issues) guide if you haven't already!

**Comment by julianguyen** [comment]
This is an interesting idea - integrating these Strategies as part of a TODO list. We could generate this TODO list from Strategies that are tagged in Moments (as these types of posts are supposed to reflect a user's current state). 

We would have to think about what kind of statuses we want on a TODO item, e.g.
* In-progress
* Completed
* Cancelled

**Comment by alekpentchev** [comment]
I agree, it's an interesting idea. Maybe we could introduce another tab to the sidemenu would be a nice place for that?
And each strategy could be automatically translated to a TODO where, as stated by you, could check / uncheck them as done.
The strategy list could show on each tab how many days the user has followed it.

**Comment by julianguyen** [comment]
I like that idea @alekpentchev! That way this TODO feature can be easily accessed.

**Comment by julianguyen** [comment]
So something we need to think about is how this progress tracking feature could potentially leverage the "daily reminder email" checkbox we have set up:

![image](https://user-images.githubusercontent.com/3010728/77861277-c7efc980-71c8-11ea-8195-dc6467df40b8.png)

If someone is "finished" a Strategy, do we disable this checkbox?

In order to work on this, @GayathriVenkatesh or @alekpentchev, we should write out a plan for what we're doing with the data.

**Comment by GayathriVenkatesh** [comment]
@julianguyen So what I originally had in mind was that every strategy can be marked as "finished" on a daily basis (or as frequently as desired by the user). This wouldn't interfere with our reminder emails because each strategy is a continuous activity and is meant to be followed regularly (for which reminders are needed). The emails will have to be explicitly disabled by the user, if he chooses to discontinue the strategy.

We can allow the user to add how often he wants to follow a strategy. Consequently, under every date (of the current week), we can display all the strategies that should've been done that day, with a checkbox. We can then leverage this data to generate weekly reports with motivating quotes, which I feel would encourage users to follow their strategies regularly.

**Comment by julianguyen** [comment]
@GayathriVenkatesh 

I like the idea of this feature not interfering with the reminder emails actually! I think through messaging, we can let users know this.

What you described sounds good to me. I think a good first start is updating the data model to reflect these changes. I think we can chat about weekly reports after, but I like the idea :)
